src/EditorFeatures/Core/Classification/Syntactic/SyntacticClassificationTaggerProvider.TagComputer.cs

```diff
@@ -272,26 +282,29 @@ private async ValueTask ProcessChangesAsync(ImmutableSegmentedList<ITextSnapshot
                 Contract.ThrowIfTrue(snapshots.IsDefault || snapshots.IsEmpty);
                 var currentSnapshot = GetLatest(snapshots);
 
-                var classificationService = TryGetClassificationService(currentSnapshot);
-                if (classificationService == null)
+                if (TryGetClassificationService(currentSnapshot) is not (var solutionServices, var classificationService))
                     return;
 
                 var currentDocument = currentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
                 if (currentDocument == null)
                     return;
 
-                var (previousSnapshot, previousDocument, previousRoot) = GetLastProcessedData();
+                SumType<SyntaxNode, Document>? previousDocumentOrRoot = null;
+                if (GetLastProcessedData() is { } data)
+                {
+                    previousDocumentOrRoot = data.lastDocumentOrRoot;
+                }
 
                 // Optionally pre-calculate the root of the doc so that it is ready to classify
                 // once GetTags is called.  Also, attempt to determine a smaller change range span
                 // for this document so that we can avoid reporting the entire document as changed.
 
                 var currentRoot = await currentDocument.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-                var changedSpan = await ComputeChangedSpanAsync().ConfigureAwait(false);
+                var changedSpan = await ComputeChangedSpanAsync(solutionServices, classificationService, previousDocumentOrRoot, currentSnapshot, currentDocument, currentRoot, _diffTimeout, cancellationToken).ConfigureAwait(false);
```

Comment:
i would really love for tehse to not be changed to static.  they were capturing local functions intentionally :)

Reply:
will return these to non-static

Reply:
➡️ Amended the commit to return these to non-static.